src/topology/inseparable.lean

```diff
@@ -178,12 +179,201 @@ def specialization_preorder : preorder X :=
 
 variable {X}
 
+local attribute [instance] specialization_preorder
+
 /-- A continuous function is monotone with respect to the specialization preorders on the domain and
 the codomain. -/
-lemma continuous.specialization_monotone (hf : continuous f) :
-  @monotone _ _ (specialization_preorder X) (specialization_preorder Y) f :=
+lemma continuous.specialization_monotone (hf : continuous f) : monotone f :=
 λ x y h, h.map hf
 
+lemma closure_singleton_eq_Iic (x : X) : closure {x} = Iic x :=
+ext $ λ _, specializes_iff_mem_closure.symm
+
+/-- A subset `S` of a topological space is stable under specialization
+if `x ∈ S → y ∈ S` for all `x ⤳ y`. -/
+def stable_under_specialization (s : set X) : Prop :=
```

Comment:
I suggest you start a new file `specialization.lean` that contains all the specialization related stuff. That also solves the "import"-issue, and it will help keep files small.